Source/objects.cpp

```diff
@@ -2548,15 +2548,17 @@ void OperateShrineEnchanted(Player &player)
 		spell = 1;
 		for (uint8_t j = static_cast<uint8_t>(SpellID::Firebolt); j < maxSpells; j++) {
 			if ((player._pMemSpells & spell) != 0 && player._pSplLvl[j] < MaxSpellLevel && j != spellToReduce) {
-				player._pSplLvl[j]++;
-				NetSendCmdParam2(true, CMD_CHANGE_SPELL_LEVEL, j, static_cast<uint8_t>(player._pSplLvl[j] + 1));
+				uint8_t newSpellLevel = static_cast<uint8_t>(player._pSplLvl[j] + 1);

```

Comment:
I left the previous static_cast intact. wasn't really sure why it's there to begin with but just wanted to make as little changes as possible to fix the issue with magic requirements. I'm going to open another PR since this PR isn't complete (forgot to refresh item cache). I will make that change in that PR.